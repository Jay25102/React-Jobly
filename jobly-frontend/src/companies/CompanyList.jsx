import React, { useState, useEffect} from "react";
import JoblyApi from "../../api";
import CompanyCard from "./CompanyCard";
import SearchForm from "../forms/SearchForm";


/**
 * Fetches list of companies from the API on load.
 * 
 * Renders a company card to display company info for each
 * company.
 */
function CompanyList() {
    const [companies, setCompanies] = useState(null);
    console.debug("CompanyList");

    useEffect(function fetchCompaniesOnMount() {
        console.debug("CompanyList useEffect fetchCompaniesOnMount");
        search();
    }, []);

    async function search(name) {
        let response = await JoblyApi.getCompanies(name);
        setCompanies(response);
    }

    if (!companies) return <div>loading...</div>

    return (
        <div className="CompanyList col-md-8 offset-md-2">
            <SearchForm searchFor={search}/>
            {companies.map(c => (
                <CompanyCard 
                    key={c.handle}
                    handle={c.handle}
                    name={c.name}
                    description={c.description}
                    logoUrl={c.logoUrl}
                />
            ))}
        </div>
    )
}

export default CompanyList;