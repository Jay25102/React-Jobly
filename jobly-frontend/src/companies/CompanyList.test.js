import React from "react";
import { render } from "@testing-library/react";
import CompanyList from "./CompanyList";

describe("CompanyList", function() {
    test("it renders without crashing", function() {
        render(<CompanyList/>);
    });
    test("it matches snapshot", function() {
        const { asFragment } = render(<CompanyList/>);
        expect(asFragment()).toMatchSnapshot();
    });
});
