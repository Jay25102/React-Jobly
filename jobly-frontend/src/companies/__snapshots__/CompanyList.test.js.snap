// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CompanyList it matches snapshot 1`] = `
<DocumentFragment>
  <div>
    loading...
  </div>
</DocumentFragment>
`;
